lib/record_store/provider/ns1.rb

```diff
@@ -139,7 +139,7 @@ def update(id, record, zone)
         )
       end
 
-      def build_from_api(api_record, zone)
+      def build_from_api(api_record, zone) # rubocop:disable Lint/UnusedMethodArgument
```

Comment:
This is not an interface here, please rename the parameter `zone` in `_zone` instead of ignoring the error!

Also, side note but, I'm surprised we don't have an error checking verifying that the `api_record['domain']` corresponds to the `zone` we received. That shouldn't happen, but if it does we just act as if nothing here ? Maybe @cttttt would have more context on this ?

Reply:
- `zone` isn't used here and this should actually a private method.  The parameter can probably be removed.  I think this was modelled off of similar private methods in other providers, which used `zone` because record domains in the API weren't fqdns.  The NS1 API returns fqdns as record domains.